tests/utils.py

```diff
@@ -760,3 +761,50 @@ def load_kasvs_ecdh_vectors(vector_data):
             }
 
     return vectors
+
+
+def load_x963_vectors(vector_data):
+    """
+    Loads data out of the X9.63 vector data
+    """
+
+    vectors = []
+
+    # Sets Metadata
+    hashname = None
+    vector = ["", 0, 0, None, 0, None, 0, None]
+    for line in vector_data:
+        line = line.strip()
+
+        if line.startswith("[SHA"):
+            hashname = line[1:-1]
+            shared_secret_len = 0
+            shared_info_len = 0
+            key_data_len = 0
+        elif line.startswith("[shared secret length"):
+            shared_secret_len = int(line[1:-1].split("=")[1].strip())
+        elif line.startswith("[SharedInfo length"):
+            shared_info_len = int(line[1:-1].split("=")[1].strip())
+        elif line.startswith("[key data length"):
+            key_data_len = int(line[1:-1].split("=")[1].strip())
+        elif line.startswith("COUNT"):
+            count = int(line.split("=")[1].strip())
+            vector[0] = hashname
+            vector[1] = count
+            vector[2] = shared_secret_len
+            vector[4] = shared_info_len
+            vector[6] = key_data_len
+        elif line.startswith("Z"):
+            vector[3] = line.split("=")[1].strip()
+            assert math.ceil(shared_secret_len / 8) * 2 == len(vector[3])
+        elif line.startswith("SharedInfo"):
+            if shared_info_len != 0:
+                vector[5] = line.split("=")[1].strip()
+                assert math.ceil(shared_info_len / 8) * 2 == len(vector[5])
+        elif line.startswith("key_data"):
+            vector[7] = line.split("=")[1].strip()
+            assert math.ceil(key_data_len / 8) * 2 == len(vector[7])
+            vectors.append(vector)
+            vector = ["", 0, 0, None, 0, None, 0, None]
```

Comment:
I find the usage of a list here very confusing when trying to read the tests. Could we switch it to a dict?